### ISSUE TITLE
Separate external interface

### ISSUE THREAD

**Issue by PipelineBaron** [RaasAhsan/pcc#6]
We should create an `include` folder next to `src`. The `include` folder will be for data types and functions that are made publicly accessible by this library. Non publicly exposed (library private) data structures and functions can still be defined in header files in the `src` folder

**Comment by RaasAhsan** [comment]
So header files (which i've restricted to public api objects) would go into include and everything else in src?

**Comment by PipelineBaron** [comment]
Yeah exactly. Creating a header file In `include` is analogous to to exporting a function in an NPM module.
